app/scripts/services/hpa.js

```diff
@@ -146,15 +168,27 @@ angular.module("openshiftConsole")
         return $q.when([]);
       }
       return MetricsService.isAvailable().then(function(metricsAvailable) {
+        var v2HPAWarning = hasV2HPAScaler(scaleTarget, hpaResources);
         return _.compact([
           hasMetricsAvailableWarning(metricsAvailable),
-          hasCPURequestWarning(scaleTarget, limitRanges, project),
+          v2HPAWarning ? v2HPAWarning : hasCPURequestWarning(scaleTarget, limitRanges, project),
           hasCompetingAutoscalersWarning(hpaResources),
           hasCompetingDCAndAutoscalerWarning(scaleTarget, hpaResources)
         ]);
       });
     };
 
+    // Get HPA warnings relevant to a Resource list page.
+    //
+    // This method returns the same set as the above getHPAWarnings but
+    // filters out the warnings that don't need to be fully surfaced to
+    // each individual resource.
+    var getHPAWarningsForResource = function(scaleTarget, hpaResources, limitRanges, project) {
+      return getHPAWarnings(scaleTarget, hpaResources, limitRanges, project).then(function(warnings) {
+        return $filter('hpaWarningsForResource')(warnings);
```

Comment:
nit: better to resolve the filter outside the loop

Reply:
I'll drop the `filter` per your other comment.